src/index.js

```diff
@@ -1,14 +1,23 @@
 import behaviorSubject from 'callbag-behavior-subject'
-import { useRef } from 'react'
+import { useRef, useEffect } from 'react'
 
 export default function useValueCallbag(value) {
-  const value$Ref = useRef(null)
+  const value$Ref = useRef()
+  const lastVal = useRef(value)
 
-  if (value$Ref.current === null) {
+  if (!value$Ref.current) {
     value$Ref.current = behaviorSubject(value)
-  } else {
-    value$Ref.current(1, value)
+    lastVal.current = value
   }
 
-  return value$Ref.current
+  const subject = value$Ref.current
+
+  useEffect(() => {
+    if (lastVal.current !== value) {
+      subject(1, value)
+      lastVal.current = value
+    }
+  }, [value, subject])
```

Comment:
Since we are doing our own comparison within the effect I think this can be safely dropped and we don't need a dependency on `subject` since it won't ever change. Any objections?

```suggestion
  })
```

Reply:
I left it there for semantic reasons (to signify that this effect should re-run on changes of `value` and `subject`), the outcome (at least for now) is the same so it is safe to remove.